src/components/organisms/EventModal/EventModal.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useMemo } from "react";
+import Modal from "react-bootstrap/Modal";
+import { VenueEvent } from "types/venues";
+import { Room } from "types/rooms";
+
+import { WithVenueId } from "utils/id";
+import { enterVenue } from "utils/url";
+import { hasEventFinished, isEventLive } from "utils/event";
+import { formatUtcSecondsRelativeToNow } from "utils/time";
+
+import { useRelatedVenues } from "hooks/useRelatedVenues";
+import { useRoom } from "hooks/useRoom";
+
+import { Button } from "components/atoms/Button";
+
+import "./EventModal.scss";
+
+const emptyRoom: Room = {
+  title: "",
+  subtitle: "",
+  url: "",
+  about: "",
+  x_percent: 0,
+  y_percent: 0,
+  width_percent: 0,
+  height_percent: 0,
+  isEnabled: false,
+  image_url: "",
+};
+
+export interface EventModalProps {
+  show: boolean;
+  onHide: () => void;
+  event: WithVenueId<VenueEvent>;
+}
+
+export const EventModal: React.FC<EventModalProps> = ({
+  event,
+  onHide,
+  show,
+}) => {
+  const { currentVenue: eventVenue } = useRelatedVenues({
+    currentVenueId: event.venueId,
+  });
+
+  const eventRoom = useMemo(
+    () =>
+      eventVenue?.rooms?.find((room) => room.title === event.room) ?? emptyRoom,
+    [eventVenue, event]
+  );
+
+  const { enterRoom } = useRoom({
+    room: eventRoom,
+    venueName: eventVenue?.name ?? "",
+  });
+
+  const eventLocationToDisplay = event.room
+    ? event.room
+    : eventVenue?.name ?? event.venueId;
+
+  const goToEventLocation = () => {
+    onHide();
+
+    if (event.room) {
+      enterRoom();
+    } else {
+      enterVenue(event.venueId);
+    }
+  };
```

Comment:
> Do we need to explicitly make this distinction here, or does useRoom's enterRoom already do it for us?

I think we do. `useRoom`s `enterRoom` navigates to the target room venue (or external url). But event can be attached to a venue without a room. 

> potentially should include isLive as a guard statement if (!isLive) return at the start as well.

I wouldn't include `isLive` because I used this function in 2 places:

![](https://user-images.githubusercontent.com/63313732/119663010-fc2c3d80-bdff-11eb-8045-1076e4e56b92.png)

1. Underlined `Campfire` - on click - navigates to the room / venue (no matter if the event is live or not)
2. `Happening now in the campfire` button - enabled only if the event is live (I make the button disabled unless `isLive`)

Reply:
RE: isLive, check with @sparkletown/product on that, as we have had issues in the past where ‘secret’/‘hidden’ venues have been discovered/accessed early through the schedule events because users could click through to them early. I expect we may want to block both button and link when it’s not live. Not 100% on that though.

Reply:
@sunny-viktoryia This seems to be the issue I was referring to: https://github.com/sparkletown/internal-sparkle-issues/issues/147

(also, for reference, there is a slack thread discussing this further at https://doesitsparkle.slack.com/archives/C01MBE28NRZ/p1622146978007700)